go-controller/pkg/ovn/egressip_test.go

```diff
@@ -3988,9 +3988,15 @@ var _ = ginkgo.Describe("OVN master EgressIP Operations", func() {
 				// 3) check to make sure the logicalPortCache is used always even if podAssignment already has the podKey
 				ginkgo.By("delete the egress IP pod and force the deletion to fail")
 				egressPod1.Annotations = map[string]string{}
-				_, err = fakeOvn.fakeClient.KubeClient.CoreV1().Pods(egressPod1.Namespace).Update(context.TODO(), &egressPod1, metav1.UpdateOptions{})
+				_, err = fakeOvn.fakeClient.KubeClient.CoreV1().Pods(egressPod1.Namespace).Update(context.TODO(), egressPod1, metav1.UpdateOptions{})
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
-				gomega.Expect(egressPod1.Annotations).To(gomega.Equal(map[string]string{}))
+				// Wait for the cleared annotations to show up client-side
+				gomega.Eventually(func() int {
```

Comment:
oh! this is interesting... it takes some finite timeline for the fake test watchers to pick this up?

Reply:
I guess it mostly depends on the load of the machine where github CI runs... I wouldn't be surprised if on a second run this test passed :)

Reply:
Yeah, there's lots of goroutines that watch various channels in the informer and our watch factory, so nothing is going to be very synchronous especially on slow systems with small numbers of cores like the GH CI runners.